app/models/concerns/profile_searching.rb

```diff
@@ -135,6 +135,18 @@ def ssg_version_search(_filter, operator, value)
                    ::Xccdf::Benchmark.where.not(version: value)
       { conditions: benchmark_id.in(benchmarks.pluck(:id)).to_sql }
     end
+
+    def profiles_for_os_major(major_os_version)
+      ::SupportedSsg.sorted_ssgs_for_os_major(major_os_version)
+                    .each_with_object({}) do |supported_ssg, latest_profiles|
+        supported_ssg.profiles.keys.each do |profile|
+          next if latest_profiles.values.flatten.include?(profile)
+
+          latest_profiles[supported_ssg.version] ||= []
+          latest_profiles[supported_ssg.version] << profile
```

Comment:
Up until now the profiles listed on SupportedSsgs were informational. This creates a dependency on the list of profile names with Profile objects. The naming convention is a bit different (names are short, the objects are exact). There might be some name mismatch, therefore if we are going with this implementation, we would need to revise the SupportedSsg list/yaml.

Other option, that I'd favor more (if it does not have a flaw), is do a union of all *supported canonical* profiles across selected major version. Those would be distinct by `ref_name` where the one with higher SSG would be returned. The sorting might be a bit trickier to do in SQL.

@skateman wdyt?

Reply:
@vkrizan this is not true, if you check [how the current endpoint works](https://github.com/RedHatInsights/compliance-backend/blob/67630b7dac18c2db7be451ce2ee94e24078f7d60/app/models/concerns/benchmark_searching.rb#L32-L39) you'll see a lot of similarities. The list yaml is generated from the datastreams, so we have no direct way to influence the content of the YAML. But for sure, we can do a check if there's any mismatch.

Reply:
> this is not true, if you check [how the current endpoint works](https://github.com/RedHatInsights/compliance-backend/blob/67630b7dac18c2db7be451ce2ee94e24078f7d60/app/models/concerns/benchmark_searching.rb#L32-L39) you'll see a lot of similarities.

@skateman indeed the version matching for supported benchmarks works with SupportedSSGs; the comment addresses the use of `SupportedSsg.profiles` ([example](https://github.com/RedHatInsights/compliance-backend/blob/4fcf793ecea42f2bf8293effff746f9bc696d098/config/supported_ssg.default.yaml#L254-L273)) which are just being used by this implementation.

> The list yaml is generated from the datastreams, so we have no direct way to influence the content of the YAML. 

Nope, the list of profiles is handcrafted manually. We get it from OpenSCAP team (by `state.yaml`).

Definitely this approach gives us more control on what is shown. The drawback, however, is the human factor.

Reply:
From slack:
```
SELECT * FROM profiles WHERE benchmark.supported = TRUE GROUP BY benchmark_version DESC;
```
pseudo-SQL :wink: 

I am leaning towards this solution as we would not need to do 3 queries on our own, but GQL can self-optimize it.